### PR TITLE
ISI-4132 | Adding defaults array to API constructor; adding accompanying tests.

### PR DESCRIPTION
**Pull request by ocaballero05**

### CODE REVIEW

File: src/SwiftReachApi/SwiftReachApi.php [hunk 1]
```diff
@@ -32,26 +32,55 @@ class SwiftReachApi
...
         );
     }
+
+    /**
+     * @return array
+     */
+    protected function getDefaultDefaults()
+    {
+        $default_defaults = [
+            "timeout"         => 30,
```
Comment (alai-illuminateed) [inline src/SwiftReachApi/SwiftReachApi.php]: can you lower this to 5 seconds

File: src/SwiftReachApi/SwiftReachApi.php [hunk 2]
```diff
@@ -32,26 +32,55 @@ class SwiftReachApi
      */
     private $base_url;
 
+    /**
+     * @var array
+     * default request options
+     */
+    private $defaults;
```
Comment (alai-illuminateed) [inline src/SwiftReachApi/SwiftReachApi.php]: I think a more descriptive name than' defaults', would be best.  Like your comment maybe 'request_options'.

**Review by alai-illuminateed — CHANGES_REQUESTED** [review]

File: src/SwiftReachApi/SwiftReachApi.php (hunk 1 above)
Comment (ocaballero05) [reply src/SwiftReachApi/SwiftReachApi.php]: sure. these are the default options, so i figured i'd leave them alone, and set it to 5 seconds on the ATD side. but those can just use the defaults of 5 if that's the case; no ATD changes needed.

File: src/SwiftReachApi/SwiftReachApi.php [hunk 3]
```diff
@@ -32,26 +32,55 @@ class SwiftReachApi
...
         $this->setBaseUrl($base_url);
+        $this->setRequestOptions($request_options);
 
         $this->guzzle_client = new Client(
             array(
-                "defaults" => array(
-                    "timeout"         => 30,
-                    "connect_timeout" => 30
-                )
+                "defaults" => $default_options
```
Comment (alai-illuminateed) [inline src/SwiftReachApi/SwiftReachApi.php:67]: i think this should be $request_options

**Review by alai-illuminateed — APPROVED** [review]

**Pull request merged**